service/history/shardContext.go

```diff
@@ -785,12 +786,52 @@ func acquireShard(shardID int, svc service.Service, shardManager persistence.Sha
 	historyMgr persistence.HistoryManager, executionMgr persistence.ExecutionManager, domainCache cache.DomainCache,
 	owner string, closeCh chan<- int, config *Config, logger bark.Logger, metricsClient metrics.Client) (ShardContext,
 	error) {
-	response, err0 := shardManager.GetShard(&persistence.GetShardRequest{ShardID: shardID})
-	if err0 != nil {
-		return nil, err0
+
+	var shardInfo *persistence.ShardInfo
+
+	retryPolicy := backoff.NewExponentialRetryPolicy(50 * time.Millisecond)
+	retryPolicy.SetMaximumInterval(time.Second)
+	retryPolicy.SetExpirationInterval(5 * time.Second)
+
+	retryPredicate := func(err error) bool {
+		if common.IsPersistenceTransientError(err) {
+			return true
+		}
+		_, ok := err.(*persistence.ShardAlreadyExistError)
```

Comment:
what is this? why is it not handled previously?

Reply:
previously the shard creation / initialization is handled at the very beginning of the history service start, and if there is any error encounter, the start will be aborted.

this PR aim to increase the resilience when one shard is not accessible